### PR TITLE
Enhancement - Native CSPRNG support and OpenSSL deprecation

### LINKED ISSUE #7: Remove support for OpenSSL random byte generation
The [`OpenSslRandomByteGenerator`](https://github.com/robinpowered/php-ntlm/blob/v0.3.4/src/Robin/Ntlm/Crypt/Random/OpenSslRandomByteGenerator.php) currently in use is _dangerous_, as it's not truly cryptographically secure. Why? Because the OpenSSL extension for PHP has some major flaws, as pointed out by @paragonie-scott in paragonie/random_compat#96:

> - [It's not fork safe](https://github.com/ramsey/uuid/issues/80)
> - [In most versions of PHP, it lies about being secure](https://bugs.php.net/bug.php?id=70014)
> - And today I learned that OpenSSL, by default (i.e. unchangable from PHP land) [uses MD5 as a CSPRNG](https://wiki.openssl.org/index.php/Random_Numbers#Generators) (thanks @atoponce)

OpenSSL has more information on [the fork safety issue on their wiki page](https://wiki.openssl.org/index.php/Random_fork-safety).

So, now that we now that it's dangerous, we have to think about the strategy for removal. The problem is that, as pointed out in #1, [the current alternative RNG is based on `mcrypt`](https://github.com/robinpowered/php-ntlm/blob/v0.3.4/src/Robin/Ntlm/Crypt/Random/McryptRandomByteGenerator.php), which is an abandoned extension and [will be removed from PHP in the future](https://wiki.php.net/rfc/mcrypt-viking-funeral). So, if we remove the OpenSSL implementation, then the only other implementation will be one that is based on abandonware. Not good.

Instead of just removing the OpenSSL support, then, we should add another alternative RNG. PHP 7 added the beautiful [`random_bytes()` function](http://php.net/manual/en/function.random-bytes.php), but this library supports PHP 5.4.0, so we couldn't just depend on that. Luckily there's a very popular and even officially recommended "polyfill" library: [**random_compat**](https://github.com/paragonie/random_compat). This polyfill backports the `random_bytes()` function to PHP 5 so that users can use the newer, safer, easier RNG without _having_ to upgrade their PHP runtime version. Since **php-ntlm** doesn't dictate which you should use, we can use [Composer's `suggest` schema feature](https://getcomposer.org/doc/04-schema.md#suggest) to suggest installing the polyfill library. We can then add documentation in the class pointing to the polyfill.

Finally, how do we go about the actual removal of the OpenSSL RNG? I'm thinking that we "deprecate" the `OpenSslRandomByteGenerator` class with a documentation notice regarding it's security implications (likely pointing to this issue). It would likely be a good idea to actually deprecate the `McryptRandomByteGenerator` class too, but have a longer life-span before final removal. That way the only real suggested RNG would be `random_bytes()`. Those details can be figured out at PR time, though.

### PR DESCRIPTION
**Pull request by Rican7**
This PR resolves #7, via deprecation. We'll out-right _remove_ the OpenSSL PRNG support at a later date/version.

As #7 described, this PR also introduces a new implementation of the `RandomByteGeneratorInterface` that utilizes PHP 7's [`random_bytes()`](http://php.net/manual/en/function.random-bytes.php) function. Also, this PR adds [`paragonie/random_compat`](https://github.com/paragonie/random_compat) as a Composer dev dependency and suggestion so that users can us a more cryptographically secure psuedo-random data generator for PHP 5 versions.

This PR also deprecates the `McryptRandomByteGenerator`, for the reasons mentioned in #7 **and** in #1.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ php:
   - 5.4
   - 5.5
   - 5.6
+  - 7.0
```
Comment (jstruzik) [inline .travis.yml:8]: Stez.

File: src/Robin/Ntlm/Crypt/Random/McryptRandomByteGenerator.php [hunk 2]
```diff
@@ -16,6 +16,9 @@
  * "mcrypt" extension.
  *
  * @link http://php.net/mcrypt
+ * @deprectated NOTE! This implementation is deprecated, as the mcrypt library
```
Comment (jstruzik) [inline src/Robin/Ntlm/Crypt/Random/McryptRandomByteGenerator.php:19]: Good idea.

**Comment by jstruzik** [comment]
LGTM :+1:

File: src/Robin/Ntlm/Crypt/Random/McryptRandomByteGenerator.php (hunk 2 above)
Comment (zachdunn) [inline src/Robin/Ntlm/Crypt/Random/McryptRandomByteGenerator.php:19]: "deprecated"

**Comment by brianmuse** [comment]
`Note! Nice work` LGTM :+1:

**Pull request merged**